services/digest_io.py

```
@@ -4,248 +4,258 @@
+            logger.error(f"No PRE-CLEANED digest at {pre_path}")
+            return None
+
+        digest = json.loads(pre_path.read_text())
         
         try:
-            cache_manager.atomic_write_json(json_path, serializable_digest, overwrite=True)
+            digest = self.enhance_with_ai(digest)
+            final_path = self.blogs_dir / target_date / f"FINAL-{target_date}_digest.json"
+            self.cache.atomic_write_json(final_path, digest, overwrite=True)
+            logger.info(f"Created FINAL digest: {final_path}")
+            return digest
```

Ruff (0.12.2)

Consider moving this statement to an `else` block

(TRY300)